### PR TITLE
Add OpenMandriva repositories

### PR DESCRIPTION
**Pull request by berolinux**
Signed-off-by: Bernhard Rosenkränzer <bero@lindev.ch>

### CODE REVIEW

File: repos.d/rpm/openmandriva.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+###########################################################################
+# OpenMandriva
+###########################################################################
+- name: openmandriva_3
+  type: repository
+  desc: OpenMandriva 3.x
+  family: openmandriva
```
Comment (AMDmi3) [inline repos.d/rpm/openmandriva.yaml]: Please add this to `families` list in `repology-schemacheck.py`, this should also fix travis build

File: repos.d/rpm/openmandriva.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+###########################################################################
+# OpenMandriva
+###########################################################################
+- name: openmandriva_3
+  type: repository
+  desc: OpenMandriva 3.x
```
Comment (AMDmi3) [inline repos.d/rpm/openmandriva.yaml]: Is it not worth to create separate repo for each 3.x? As I see, there's only 3.0 currently, but may 3.1 also appear?

File: repos.d/rpm/openmandriva.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  family: openmandriva
+  color: '045e9d'
+  sources:
+      # XXX: */backports_testing and */updates_testing not included, are these needed?
```
Comment (AMDmi3) [inline repos.d/rpm/openmandriva.yaml]: There are no `*/backports_testing and */updates_testing`, please remove comment.

File: repos.d/rpm/openmandriva.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  tags: [ all, production, openmandriva, rpm ]
+
+- name: openmandriva_cooker
+  type: repository
+  desc: OpenMandriva Cooker
+  family: openmandriva
+  color: '045e9d'
+  sources:
+      # XXX: */backports_testing and */updates_testing not included, are these needed?
+    - name: [ main/release, main/updates, main/testing, non-free/release, non-free/updates, non-free/testing, restricted/release, restricted/updates, restricted/testing ]
```
Comment (AMDmi3) [inline repos.d/rpm/openmandriva.yaml]: Most of these subrepos are not there, there's only */release. Same question for contrib repo.

File: repos.d/rpm/openmandriva.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+# OpenMandriva
+###########################################################################
+- name: openmandriva_3
+  type: repository
+  desc: OpenMandriva 3.x
+  family: openmandriva
+  color: '045e9d'
+  sources:
+      # XXX: */backports_testing and */updates_testing not included, are these needed?
+    - name: [ main/release, main/updates, main/testing, non-free/release, non-free/updates, non-free/testing, restricted/release, restricted/updates, restricted/testing ]
```
Comment (AMDmi3) [inline repos.d/rpm/openmandriva.yaml]: Isn't `contrib` repo worth adding?

File: repos.d/rpm/openmandriva.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      fetcher: Repodata
+      parser: Repodata
+      url: 'http://abf-downloads.openmandriva.org/3.0/repository/SRPMS/{source}/'
+      subrepo: '{source}'
+  repolinks:
+    - desc: OpenMandriva home
+      url: http://openmandriva.org/
```
Comment (AMDmi3) [inline repos.d/rpm/openmandriva.yaml]: https://

File: repos.d/rpm/openmandriva.yaml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  sources:
+      # XXX: */backports_testing and */updates_testing not included, are these needed?
+    - name: [ main/release, main/updates, main/testing, non-free/release, non-free/updates, non-free/testing, restricted/release, restricted/updates, restricted/testing ]
+      fetcher: Repodata
+      parser: Repodata
+      url: 'http://abf-downloads.openmandriva.org/cooker/repository/SRPMS/{source}/'
+      subrepo: '{source}'
+  repolinks:
+    - desc: OpenMandriva home
+      url: http://www.openmandriva.org/
```
Comment (AMDmi3) [inline repos.d/rpm/openmandriva.yaml]: https://

**Comment by berolinux** [comment]
Thanks, fixed in updated patch.

For the 3.x bit, it's actually meant the way it is now. There won't be a 3.1 release, but there have been 3.01 and 3.02 releases (and there will be 3.03 and probably 3.04), but all of those are just respins of 3.0 with everything in updates/ included -- so installing 3.0 and then installing all updates is exactly the same as installing 3.02.

**Comment by AMDmi3** [comment]
It turns out that `repodata` subdirectory only is only present in contrib/release and main/release for both repos. We'll only be able to parse these subrepos.

**Comment by AMDmi3** [comment]
Also the 3.x and cooker repo contents seem to be completely the same.

**Comment by berolinux** [comment]
That's a temporary issue with the infrastructure, we're currently in the process of switching from the old style Mandrake metadata (see e.g. http://abf-downloads.openmandriva.org/cooker/repository/SRPMS/main/release/media_info/ ) to "standard" repodata.

Both cooker metadata being out of sync and missing metadata for updates, restricted, non-free etc. should be fixed soon.

**Comment by AMDmi3** [comment]
Okay, then it's good. I'll add some required changes (leave only repos which have repodata for now, add required global rules, teach repodata parser to decompress .xz etc.) and merge this. Please submit an issue/PR when more repositories are converted to repodata and may thus be added to repos.d.

**Comment by AMDmi3** [comment]
It's [available](https://repology.org/statistics#openmandriva_3) on the site.

**Pull request merged**